src/Sarif.Sarifer/Rule.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.CodeAnalysis.Sarif.Sarifer
+{
+    internal class Rule
+    {
+        public string Id { get; set; }
```

Comment:
>set;  [](start = 32, length = 5)

Since these are all initialized in the ctor and we don't expect them to change, make them `get`-only.